dex/src/main/scala/com/wavesplatform/dex/Matcher.scala

```diff
@@ -340,29 +341,29 @@ class Matcher(settings: MatcherSettings)(implicit val actorSystem: ActorSystem)
     ) unsafeTap sendBalanceChanges(settings.wavesBlockchainClient.balanceStreamBufferSize)
   }
 
+  private def validateForAddress(ao: AcceptedOrder, tradableBalance: Map[Asset, Long]): Future[Either[MatcherError, Unit]] = {
```

Comment:
What about `FutureResult[Unit]` type instead of this monster: `Future[Either[MatcherError, Unit]]`?

Reply:
It's not possible, because `FutureResult[Unit] = EitherT[Future, MatcherError, Unit] != Future[Either[MatcherError, Unit]]` which is required by `AddressActor`.
Also it is a good practice to not expose `EitherT` outside, otherwise we have to transform all futures (in our case) into monad transformers. In addition if there are multiple types of monad transformers (e.g. EitherT and OptionT) you have to deal with much complexity, because they doesn't compose together. In short, let's try to keep the principle of least power if it is possible.